tests/net/lib/coap/src/main.c

```diff
@@ -523,6 +523,124 @@ static int test_parse_malformed_marker(void)
 	return result;
```

Comment:
I noticed that the coap unit test is one of the few remaining network unit tests that is still not using ztest APIs. Legacy TCP test is the other one but we will remove those old TCP tests for 2.6.
I think we should convert the coap tests to use the ztest at some point, I will create an enhancement for this so that it is not forgotten.

Reply:
> Legacy TCP test is the other one but we will remove those old TCP tests for 2.6.

What do you mean by "remove tests"? Please don't remove tests, port them.